### ISSUE TITLE
Diagrams Generation

### ISSUE THREAD

**Issue by rtkelly13** [timlrx/tailwind-nextjs-starter-blog#394]
I'm always looking for ways to automatically generate diagrams in blog posts using DSLs rather than WYSIWYG solutions.

I've read this blog post from GitHub where they have integrated a really nice looking solution for integrating diagram generation from MD files 
https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

There is a plugin for remark which seems like it could integrate easily.
https://github.com/sjwall/mdx-mermaid

I've already considered a solution using draw.io integrated into VSCode using SVG diagrams. This was in a previous issue on this repo as a good solution to solve this issue.

Would you be interested in me trying this out an possibly posing a PR if it works well enough?

**Comment by aloisdg** [comment]
I prefer PlantUML to Mermaid, but Mermaid is better than nothing.

https://wyssmann.com/blog/2021/03/diagrams-as-code-plantuml-or-mermaid/

**Comment by zlwu** [comment]
> I don't really like the approach taken by `mdx-mermaid`. It basically injects mermaid.js as a dependency and renders the markup as a SVG on the client-side.
> 
> There's a [discussion on remarksjs](https://github.com/remarkjs/remark/discussions/932) about adding mermaid support as well. Given that they have just introduced it on Github, I will just wait for the core maintainers of remarkjs to decide how best to handle it and not accept a PR till then.
> 
> There's a related thread here as well: [remcohaszing/remark-mermaidjs#3](https://github.com/remcohaszing/remark-mermaidjs/issues/3)

Maybe we can use [mermaid-cli](https://github.com/mermaid-js/mermaid-cli) to compile *.mmd file (mermaid source file) to svg directly.

**Comment by danielmackay** [comment]
Any update on this?  I would love to be able to render Mermaid diagrams natively within the template.

**Comment by abernier** [comment]
https://github.com/timlrx/tailwind-nextjs-starter-blog/pull/971

**Comment by awaisshah228** [comment]
For anyone looking for a diagram generation tool beyond Mermaid — [Calcstack](https://calcstack.co/editor) supports an Eraser.io-style DSL for diagram-as-code with AI generation. You can describe diagrams in plain English or use the DSL syntax, and export as PNG/SVG/PDF to embed in blog posts. It supports flowcharts, architecture diagrams, sequence diagrams, ERDs, mind maps, and more with 1000+ icons and ELK-based auto-layout.
